release/release.go

```diff
@@ -1297,3 +1353,181 @@ func linuxRelease(v version.Version) {
 func canPerformStableRelease(v version.Version) bool {
 	return v.IsStable() && env.EvgIsTagTriggered()
 }
+
+func downloadMongodAndShell(v string) {
+	err := os.Mkdir("bin", 0700)
+	if err != nil && !os.IsExist(err) {
+		check(err, "create bin dir")
+	}
+
+	pf, err := platform.GetFromEnv()
+	check(err, "get platform")
+
+	feedURL := "http://downloads.mongodb.org/full.json"
+
+	var feed download.ServerJSONFeed
+
+	res, err := http.Get(feedURL)
+	check(err, "get the server JSON feed")
+
+	err = json.NewDecoder(res.Body).Decode(&feed)
+	check(err, "decode JSON feed")
+
+	url, githash, serverVersion, err := feed.FindURLHashAndVersion(v, pf.Name, string(pf.Arch), "enterprise")
+	check(err, "get URL from JSON feed")
+
+	fmt.Printf("URL: %v\n", url)
+	fmt.Printf("GitHash: %v\n", githash)
+	fmt.Printf("Version: %v\n", serverVersion)
+
+	downloadBinaries(url)
+	if semver.Compare(fmt.Sprintf("v%s", serverVersion), "v6.0.0") > -1 {
+		// serverVersion >= 6.0.0
+		downloadShell(serverVersion)
+	}
+}
+
+func downloadBinaries(url string) {
+	tempDir, err := os.MkdirTemp("bin", "")
+	check(err, "create temp dir")
+
+	filename := filepath.Base(url)
+	tempPath := filepath.Join(tempDir, filename)
+	packageFile, err := os.Create(tempPath)
+	check(err, "create the server package file")
+
+	res, err := http.Get(url)
+	check(err, "get the server package")
+
+	_, err = io.Copy(packageFile, res.Body)
+	check(err, "write server package file")
+
+	fmt.Printf("extension: %v\n", filepath.Ext(filename))
+
+	switch filepath.Ext(filename) {
+	case ".zip":
+		fmt.Printf("extracting to: %v\n", tempDir)
+		unzip(tempPath, tempDir)
+	case ".tgz":
+		fmt.Printf("extracting to: %v\n", tempDir)
+		untargz(tempPath, tempDir)
```

Comment:
Maybe we can add a case "default" and error out.

Reply:
Good idea